### PR TITLE
Add support for React in Kahuna

### PR DESCRIPTION
**Pull request by andrew-nowak**
Follows #3608, #3609 

## What does this change?

Typescript compiler from #3609 supports compiling jsx without further changes, so almost no further configuration is needed to be able to add React components to grid. 

Migrate the description too short warning on the upload page to React (it's nice and straightforward).

## How can success be measured?

Does the upload page's description warning look and the same as in the angular implementation? 

## Tested? Documented?
- [x] locally by committer
- [ ] locally by Guardian reviewer
- [ ] on the Guardian's TEST environment
- [ ] relevant documentation added or amended (if needed)

### CODE REVIEW

File: kahuna/package.json [hunk 1]
```diff
@@ -23,6 +23,10 @@
     "pandular": "0.1.6",
     "pikaday": "^1.8.2",
     "pikaday-time": "^1.6.1",
+    "preact": "^10.6.5",
```
Comment (rhystmills) [inline kahuna/package.json:26]: Do we need preact?

File: kahuna/public/js/components/gr-description-warning/gr-description-warning.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+interface GrDescriptionWarningProps {
+  description: string | undefined;
+}
+const GrDescriptionWarning: React.FC<GrDescriptionWarningProps> = ({ description }) => {
+  const showWarning = description ? shouldShowWarning(description) : true;
+
+  return showWarning && (
+    <div className={`flex-right text-small ${styles['gr-description-warning']}`}>
+      <span className={`${styles.message}`}>
+        Your description is too short! Ideally, please state who, what where, when and why.
```
Comment (rhystmills) [inline kahuna/public/js/components/gr-description-warning/gr-description-warning.tsx:24]:
```suggestion
        Your description is too short! Ideally, please state who, what, where, when and why.
```
Fix for pre-existing typo

**Review by rhystmills — APPROVED** [review]
Working nicely, have commented an opportunity to fix a historic typo. Also - the react text seems to be left aligned, while the old text was right-aligned - not a huge issue as far as I'm concerned, but perhaps something to fix if you want 100% consistency.

File: kahuna/public/js/components/gr-description-warning/gr-description-warning.module.css [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.gr-description-warning {
+  display: block;
```
Comment (andrew-nowak) [inline kahuna/public/js/components/gr-description-warning/gr-description-warning.module.css:2]:
```suggestion
```

**Pull request merged**